config/ConfigParse.cpp

```diff
@@ -0,0 +1,229 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ConfigParse.cpp                                    :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: siun <siun@student.42.fr>                  +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/05/17 22:37:44 by siun              #+#    #+#             */
+/*   Updated: 2024/05/23 18:58:56 by siun             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "ConfigParse.hpp"
+
+std::string openFile(std::string path)
+{
+	std::ifstream	file(path.c_str());
+	if (!file.is_open())
+	{
+		throw std::runtime_error("Could not open file " + path);
+	}
+	std::string	config((std::istreambuf_iterator<char>(file))
+		, std::istreambuf_iterator<char>());
+	file.close();
+	return config;
+}
+
+std::vector<std::pair<std::string, int> >	findIndent(std::string str)
```

Comment:
A vector storing pairs, Looks like using a std::map would be intuitive here.

Reply:
It should be vector. Because indent and lines are has a context of chunck with its order, and if I use map, parsing will be impossible.